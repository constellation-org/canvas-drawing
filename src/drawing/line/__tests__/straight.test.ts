import StraightLine from '../straight'

describe('[StraightLine]', () => {
  let canvas: HTMLCanvasElement
  let context: CanvasRenderingContext2D
  beforeEach(function () {
    canvas = document.createElement('canvas')
    canvas.setAttribute('width', '100')
    canvas.setAttribute('height', '100')
    context = canvas.getContext('2d') as CanvasRenderingContext2D
  })
  it('should draw a straight line', () => {
    const line = new StraightLine({
      from: {
        x: 10,
        y: 10,
      },
      to: {
        x: 90,
        y: 90,
      }
    })
    line.draw(context)
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
})
