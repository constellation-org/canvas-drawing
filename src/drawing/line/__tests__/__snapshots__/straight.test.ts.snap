// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[StraightLine] should draw a straight line 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAndJREFUeF7t3VtO5EAMheHDkpgNsCV4YmYFMzzBltgAs6VRDyBuTeKkyo7L9ee1rdg5X1dLLVnKhbhSJXCRahqGESDJvgSAAJIsgWTjcEIASZZAsnE4IYODXEl6TPYMpcbZckL+SLqVdCPpoVQKiR7GCvJT0u93c4PihGgBuZT0dKY/KA4oFpBT22tJ96A4CHy6pRUEFH+L/x22gIASgLIVBBRnlD0goDii7AUBxQmlBQQUB5RWEFA6o/QAAaUjSi8QUDqh9AQBpQNKbxBQGlE8QEBpQPECAWUniicIKDtQvEFA2YgSAQLKBpQoEFCMKJEgoBhQokFAWUE5AgSUBZSjQED5BuVIEFDOoBwNAsonlAwgoLxDyQICygtKJhBQdizKGf7aNJdMvbaa7YS8ak6LkhVk2p+vzCBTomQHmQ5lBJCpUEYBmQZlJJApUEYDKY8yIkhplFFByqKMDFISZXSQcigVQEqhVAEpg1IJpARKNZDhUSqCDI1SFWRYlMogQ6JUBxkOZQaQoVBmARkGZSaQIVBmA0mPMiNIapRZQdKizAySEmV2kHQogDxvE6fZJQbkbVk/BQogbyApTgogH0EORwHkK8ihKICcBzkMBZDvQQ5BAWQZJBwFkHWQUBRAbCBhKIDYQUJQANkG4o4CyHYQVxRA9oG4oQCyH8QFBZA2kO4ogLSDdEUBpA9INxRA+oF0QQGkL0gzCiD9QZZQfkj6u9QSEB+Qcyi/JJ3eSb94AbKWUNvnr4sTd5JO76RfvQBZjai54ErSo/UugFiTCqoDJChoaxtArEkF1QESFLS1DSDWpILqAAkK2toGEGtSQXWABAVtbfMPM22mZVxf0xEAAAAASUVORK5CYII="`;
